client/extensions/woocommerce/woocommerce-services/components/shipping-classes-field/index.js

```diff
@@ -0,0 +1,93 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { uniqBy } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import TokenField from 'components/token-field';
+import FieldError from '../field-error';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
+
+export default class ShippingClassesField extends React.Component {
+	static propTypes = {
+		id: PropTypes.string.isRequired,
+		title: PropTypes.string,
+		description: PropTypes.string,
+		value: PropTypes.array.isRequired,
+		updateValue: PropTypes.func,
+		error: PropTypes.oneOfType( [ PropTypes.string, PropTypes.bool ] ),
+		className: PropTypes.string,
+		options: PropTypes.array,
+	};
+
+	render() {
+		const { id, title, description, value, placeholder, error, className, options } = this.props;
+
+		return (
+			<FormFieldset className={ className }>
+				<FormLabel htmlFor={ id }>{ title }</FormLabel>
+				<TokenField
+					id={ id }
+					name={ id }
+					placeholder={ placeholder }
+					value={ value.map( this.getSlugFromId.bind( this ) ) }
```

Comment:
Why use slugs for the value instead of the ids? (Since we're saving the id, not the slug)

Also - Let's do the binding in the constructor like other Calypso components ([#](https://github.com/Automattic/wp-calypso/blob/master/client/gutenberg/extensions/markdown/jetpack-markdown-block-editor.jsx), [#](https://github.com/Automattic/wp-calypso/blob/master/client/lib/data-poller/poller.js), [#](https://github.com/Automattic/wp-calypso/blob/master/client/post-editor/editor-permalink/index.jsx)).